extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeCommentsExtractor.java

```diff
@@ -0,0 +1,266 @@
+package org.schabi.newpipe.extractor.services.youtube.extractors;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.schabi.newpipe.extractor.DownloadResponse;
+import org.schabi.newpipe.extractor.Downloader;
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.StreamingService;
+import org.schabi.newpipe.extractor.comments.CommentsExtractor;
+import org.schabi.newpipe.extractor.comments.CommentsInfoItem;
+import org.schabi.newpipe.extractor.comments.CommentsInfoItemExtractor;
+import org.schabi.newpipe.extractor.comments.CommentsInfoItemsCollector;
+import org.schabi.newpipe.extractor.exceptions.ExtractionException;
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.exceptions.ReCaptchaException;
+import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class YoutubeCommentsExtractor extends CommentsExtractor {
+
+    private static final String USER_AGENT = "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:61.0) Gecko/20100101 Firefox/61.0";
+
+    private List<String> cookies;
+    private String sessionToken;
+    private String title;
+    private InfoItemsPage<CommentsInfoItem> initPage;
+
+    private ObjectMapper mapper = new ObjectMapper();
+
+    public YoutubeCommentsExtractor(StreamingService service, ListLinkHandler uiHandler) {
+        super(service, uiHandler);
+    }
+
+    @Override
+    public InfoItemsPage<CommentsInfoItem> getInitialPage() throws IOException, ExtractionException {
+        // initial page does not load any comments but is required to get session token
+        // and cookies
+        super.fetchPage();
+        return initPage;
+    }
+
+    // isn't this method redundant. you can just call getnextpage on getInitialPage
+    @Override
+    public String getNextPageUrl() throws IOException, ExtractionException {
+        // initial page does not load any comments but is required to get session token
+        // and cookies
+        super.fetchPage();
+        return initPage.getNextPageUrl();
+    }
+
+    private String getNextPageUrl(JsonNode ajaxJson) throws IOException, ExtractionException {
+        Optional<JsonNode> element = Optional.ofNullable(ajaxJson.findValue("itemSectionContinuation"))
```

Comment:
Ok I understand what you are doing here. [I did it in my way too.](https://github.com/FlorianSteenbuck/OtherArray) Nothing wrong with it but a little bit overuse of new java versions and we already got json parsing and want to not depend too much on other projects.

`findValue` could also be implemented for every `Map<String, ?>` I would suggest to create a util for that and other methods you want to use from your json parsing libary.
`Optional` is here only needed for skip a null check and map a direct and indirectly with `findValue`.

- Remove `Optional`
- Decide if you want to use direct or indirect (`findValue`) mapping
- Do a null check that return ""
- Do your mappings indirect (`findValue`) or direct with the supported NanoJson Interface `Map<String, ?>`

@theScrabi I know it is not a cool Idea too implement array transformations on only arrays I should make Collections working with the functions too. The idea is too also provider a wide range of interfaces later in Javascript, Python and Lua PluginStreamingServices too